#!/usr/bin/env python
PACKAGE = "rotation_filter"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("passthrough_x_p",    double_t,    0, "PassThrough_X_Positive", 0.5,  0.0, 0.8)
gen.add("passthrough_x_n",    double_t,    0, "PassThrough_X_Negative", -0.5,  -0.8, 0.0)
gen.add("passthrough_y_p",    double_t,    0, "PassThrough_Y_Positive", 0.5,  0.0, 0.8)
gen.add("passthrough_y_n",    double_t,    0, "PassThrough_Y_Negative", -0.5,  -0.8, 0.0)
gen.add("passthrough_z_p",    double_t,    0, "PassThrough_Z_Positive", 0.5,  0.0, 0.8)
gen.add("passthrough_z_n",    double_t,    0, "PassThrough_Z_Negative", -0.5,  -0.8, 0.0)

gen.add("rotation_x",    double_t,    0, "Rotation_x", 0.0,  -180.0, 180.0)
gen.add("rotation_y",    double_t,    0, "Rotation_y", 0.0,  -180.0, 180.0)
gen.add("rotation_z",    double_t,    0, "Rotation_z", 0.0,  -180.0, 180.0)

exit(gen.generate(PACKAGE, "rotation_filter", "rotation"))
